### PR TITLE
Chime SDK for Messaging SESSION_ESTABLISH event on websocket indicates connection successful

### PR DESCRIPTION
**Pull request by dpwspoon**
Chime SDK for Messaging SESSION_ESTABLISH event on websocket indicates connection successful

**Issue #:**

**Description of changes:**
Listen to SESSION_ESTABLISH message for successful connection.  This event indicates all wiring on backend has completed and all channels the app instance user who is connected is then guaranteed to get all messages

**Testing**

1. Have you successfully run `npm run build:release` locally?'
No.  It fails with `FAIL: Requires CHANGELOG, but no CHANGELOG.md entries found since origin/master.` and I see no documentation on what to do.

2. How did you test these changes?

 - minor update to unit tests. 
 - npm install, 
 - npm test, 
 - pulled in as dependency to chat demo in https://github.com/aws-samples/amazon-chime-sdk, launched demo and confirmed work

3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?
yes, chat demo in  https://github.com/aws-samples/amazon-chime-sdk

4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?

no, this fixes the implementation on connect on websocket as it could have indicated success and in fact not be fully wired up in backend on rare race conditions.  

5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?

n/a

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ltrung** [comment]
> No. It fails with FAIL: Requires CHANGELOG, but no CHANGELOG.md entries found since origin/master. and I see no documentation on what to do.
Not sure what you meant but the CHANGELOG.md is here https://github.com/aws/amazon-chime-sdk-js/blob/master/CHANGELOG.md

File: package-lock.json [hunk 1]
```diff
@@ -1,4659 +1,8 @@
 {
   "name": "amazon-chime-sdk-js",
   "version": "2.12.0",
-  "lockfileVersion": 2,
```
Comment (ltrung) [inline package-lock.json:4]: Hmm could you not commit this change? It reverts the lockfile back to v1. Alternatively, please use Node 16 :)

**Comment by ltrung** [comment]
Could you clone and resubmit the PR? Currently, you cannot run integration test if you submit the PR from fork.

**Comment by dpwspoon** [comment]
Sure moved here: https://github.com/aws/amazon-chime-sdk-js/pull/1399.

Looks like suppose to manually update changelog (saw in recent commits), that allows it to build

**Pull request closed without merging**